app/utils/tests_import.py

```diff
@@ -450,38 +449,49 @@ def import_test_case(
     if isinstance(json_obj, types.DictionaryType):
         j_get = json_obj.get
         json_suite_id = j_get(models.TEST_SUITE_ID_KEY, None)
+        suite_oid = bson.objectid.ObjectId(suite_id)
 
         json_obj[models.TEST_SUITE_NAME_KEY] = suite_name
         if not json_suite_id:
             # Inject the suite_id value into the data structure.
             json_obj[models.TEST_SUITE_ID_KEY] = suite_id
         else:
             if json_suite_id == str(suite_id):
-                json_obj[models.TEST_SUITE_ID_KEY] = suite_id
+                json_obj[models.TEST_SUITE_ID_KEY] = suite_oid
             else:
                 utils.LOG.warning(
                     "Test suite ID does not match the provided one")
                 # XXX For now, force the suite_id value.
-                json_obj[models.TEST_SUITE_ID_KEY] = suite_id
+                json_obj[models.TEST_SUITE_ID_KEY] = suite_oid
 
         try:
             test_name = j_get(models.NAME_KEY, None)
+            set_id = j_get(models.TEST_SET_ID_KEY, None)
+            if set_id:
+                json_obj[models.TEST_SET_ID_KEY] = bson.objectid.ObjectId(set_id)
+            else:
```

Comment:
Hmmm, weird, because soon after that code a `TestCaseDocument` object is created and if nothing is in the JSON object it will default to `None`: it means it is being pushed as an empty string.